src/mlpack/core/data/image_info.hpp

```diff
@@ -44,7 +41,6 @@ namespace data {
 inline bool ImageFormatSupported(const std::string& fileName,
                                  const bool save = false);
 
-#endif

```

Comment:
If this is only needed when `HAS_STB` is present, you could move it to be a utility function in the .cpp file that you made.

Reply:
I think my approach to this problem was slightly off. I'll take a look at it once again properly. Thank you for all the advice